sdk/core/core/src/az_span.c

```diff
@@ -646,10 +646,24 @@ AZ_NODISCARD az_result _az_span_url_encode(az_span destination, az_span source,
   int32_t result_size = 0;
   for (int32_t i = 0; i < source_size; ++i)
   {
-    result_size += _az_span_url_should_encode(az_span_ptr(source)[i]) ? 3 : 1;
+    if (_az_span_url_should_encode(az_span_ptr(source)[i]))
+    {
+      if (result_size <= INT32_MAX - 3)
+      {
+        result_size += 3;
+        continue;
+      }
+    }
+    else if (result_size < INT32_MAX)
+    {
+      result_size += 1;
+      continue;
+    }
+    
+    result_size = -1;
   }
 
-  if (az_span_size(destination) < result_size)
+  if (result_size == -1 || az_span_size(destination) < result_size)
```

Comment:
Add unit tests that capture the overflow conditions so we have branch coverage and confidence this fixes the underlying issue.

Reply:
It turns out, we don't have a test for az_span_is_valid at all, I have to wand will write it.

Reply:
I meant specifically about the encode url method. For example, encoding a buffer that is larger than INT_MAX/3 in size.

Reply:
I understand, but still, I don't want to give up. Check out the new version, it does not require all that :)

Reply:
No problem now.